x/compliance/keeper/msg_server_delete_compliance_info.go

```diff
@@ -0,0 +1,65 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	dclcompltypes "github.com/zigbee-alliance/distributed-compliance-ledger/types/compliance"
+	"github.com/zigbee-alliance/distributed-compliance-ledger/x/compliance/types"
+	dclauthtypes "github.com/zigbee-alliance/distributed-compliance-ledger/x/dclauth/types"
+)
+
+func (k msgServer) DeleteComplianceInfo(goCtx context.Context, msg *types.MsgDeleteComplianceInfo) (*types.MsgDeleteComplianceInfoResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	signerAddr, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "Invalid Address: (%s)", err)
+	}
+
+	// check if sender has enough rights to delete model
+	// sender must have CertificationCenter role to certify/revoke model
+	if !k.dclauthKeeper.HasRole(ctx, signerAddr, dclauthtypes.CertificationCenter) {
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized,
+			"MsgDeleteComplianceInfo transaction should be signed by an account with the %s role",
+			dclauthtypes.CertificationCenter,
+		)
+	}
+
+	complianceInfo, found := k.GetComplianceInfo(ctx, msg.Vid, msg.Pid, msg.SoftwareVersion, msg.CertificationType)
+	if !found {
+		return nil, types.NewErrComplianceInfoDoesNotExist(msg.Vid, msg.Pid, msg.SoftwareVersion, msg.CertificationType)
+	}
+
+	// remove compliance info from the entity Device Compliance Info
+	deviceSoftwareCompliance, found := k.GetDeviceSoftwareCompliance(ctx, complianceInfo.CDCertificateId)
+	if found {
+		index, found := deviceSoftwareCompliance.IsComplianceInfoExist(msg.Vid, msg.Pid, msg.SoftwareVersion)
+		if found {
+			deviceSoftwareCompliance.RemoveComplianceInfo(index)
+		}
+	}
+	k.SetDeviceSoftwareCompliance(ctx, deviceSoftwareCompliance)
```

Comment:
I think it can be called only if `if len(deviceSoftwareCompliance.ComplianceInfo) != 0`